lib/deep_delete.rb

```diff
@@ -0,0 +1,186 @@
+# USAGE:
+#
+#   toggle_console_sql_logging
+#   DeepDelete.delete(ExtManagementSystem.find(2))
+#   DeepDelete.vacuum # to update indexes
+#
+# currently only tested on ContainerManager / ems
+class DeepDelete
```

Comment:
My only concern that's left is that this class name is a top-level constant, and I'm concerned we might pull in a gem later or something that also defines this constant (it comes close now to `DeepMerge` for example).

Thinking we should namespace this under ManageIQ, so ManageIQ::DeepDelete

Reply:
I'm thinking that it supports `:dependency => :delete` / destroy.

so deep delete seems to ring something for me.